src/components/OrderPanel/BookingDatesForm/BookingDatesForm.js

```diff
@@ -598,8 +704,57 @@ export const BookingDatesFormComponent = props => {
               onClose={() => {
                 setCurrentMonth(startDate || endDate || startOfToday);
               }}
+              onChange={values => {
+                const { startDate, endDate } = values || {};
+                const parsedStart = startDate
+                  ? getStartOf(timeOfDayFromLocalToTimeZone(startDate, timeZone), 'day', timeZone)
+                  : startDate;
+                const parsedEnd = endDate
+                  ? getStartOf(timeOfDayFromLocalToTimeZone(endDate, timeZone), 'day', timeZone)
+                  : endDate;
+                const endDateForAPI =
+                  parsedEnd && isDaily ? getExclusiveEndDate(parsedEnd, timeZone) : parsedEnd;
```

Comment:
Perhaps this could be extracted into a function, since it's now used on 2 places.

`const getStartAndEndOnTimeZone = (startDate, endDate, isDaily, timeZone) => //etc.`